src/Tools/VMC/EventLibraryInterface.cxx

```diff
@@ -0,0 +1,261 @@
+
+  if(!init_state.Tgt().IsNucleus()){
+    LOG("ELI", pINFO) << "Skippping non-nuclear target " << init_state;
+    return 0;
+  }
+
+  const int tgt_pdgc = init_state.TgtPdg();
+
+  const ProcessInfo& proc = interaction->ProcInfo();
+
+  if(!proc.IsWeakCC() && !proc.IsWeakNC()){
+    LOG("ELI", pINFO) << "Skipping unknown process " << proc;
+    return 0;
+  }
+
+  int probe_pdgc = init_state.ProbePdg();
+
+  // Use nu_mu for NC by convention
+  if(proc.IsWeakNC()){
+    if(probe_pdgc > 0) probe_pdgc = +14; else probe_pdgc = -14;
+  }
+
+  const Key key(tgt_pdgc, probe_pdgc, proc.IsWeakCC());
+
+  const auto rec_it = fRecords.find(key);
+
+  if(rec_it == fRecords.end()){
+    LOG("ELI", pINFO) << "Skippping " << key << " -- not found in library";
+    return 0;
+  }
+
+  const Record* rec = rec_it->second->GetRecord(probe_E);
+
+  if(!rec){
+    LOG("ELI", pINFO) << "Skippping " << key << " at " << probe_E << " GeV -- not found in library";
+    return 0;
+  }
+
+  return rec;
+}
+
+//____________________________________________________________________________
+std::vector<TVector3> EventLibraryInterface::Basis(const TVector3& z) const
+{
+  TVector3 up(0, 1, 0);
+  if(up.Dot(z) == 0) up = TVector3(1, 0, 0);
+
+  const TVector3 x = up.Cross(z).Unit(); // Perpendicular to neutrino and up
+  const TVector3 y = x.Cross(z).Unit();  // Defines the third axis
+
+  const double a = RandomGen::Instance()->RndEvg().Uniform(0, 2*M_PI);
+
+  const TVector3 xp =  cos(a) * x + sin(a) * y;
+  const TVector3 yp = -sin(a) * x + cos(a) * y;
+  const TVector3 zp = z.Unit();
+
+  return {xp, yp, zp};
+}
+
+//____________________________________________________________________________
+void EventLibraryInterface::Configure(const Registry & config)
+{
+  Algorithm::Configure(config);
+}
+//___________________________________________________________________________
+void EventLibraryInterface::Configure(string config)
+{
+  Algorithm::Configure(config);
+}
+
+//___________________________________________________________________________
+void EventLibraryInterface::LoadRecords() const
+{
+  assert(!fRecordFile);
+
+  std::string libPath;
+  if(!GetParamDef( "LibraryPath", libPath, std::string() )){
+    LOG("ELI", pFATAL) << "Must specify 'LibraryPath'";
+    exit(1);
+  }
+
+  Expand(libPath);
+
+  bool onDemand;
+  GetParamDef( "OnDemand", onDemand, true );
+
+  PDGLibrary* pdglib = PDGLibrary::Instance();
+
+  fRecordFile = new TFile(libPath.c_str());
+  if(fRecordFile->IsZombie()) exit(1);
+
+  TIter next(fRecordFile->GetListOfKeys());
+  while(TObject* dir = next()){
+    const std::string& tgtName = dir->GetName();
+    const TParticlePDG* tgtPart = pdglib->DBase()->GetParticle(tgtName.c_str());
+    if(!tgtPart){
+      LOG("ELI", pWARN) << "Unknown nucleus " << tgtName
+                        << " found in " << libPath
+                        << " -- skipping";
+      continue;
+    }
+
+    for(int sign: {+1, -1}){
```

Comment:
Let's use the constants we defined, not hard coded numbers

Reply:
sorry to be slow, which constants exactly?

Reply:
Done